packages/clerk-js/src/ui/components/SessionTasks/tasks/TaskSelectOrganization/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { withCardStateProvider } from '@/ui/elements/contexts';
+import { Header } from '@/ui/elements/Header';
+import { useMultipleSessions } from '@/ui/hooks/useMultipleSessions';
+import { useOrganizationListInView } from '@/ui/hooks/useOrganizationListInView';
+
+import { withTaskGuard } from '../withTaskGuard';
+import { CreateOrganizationScreen } from './CreateOrganizationScreen';
+import { SelectOrganizationScreen } from './SelectOrganizationScreen';
+
+const TaskSelectOrganizationInternal = () => {
+  const { signOut } = useClerk();
+  const { user } = useUser();
+  const { session } = useSession();
+  const { userMemberships, userSuggestions, userInvitations } = useOrganizationListInView();
+  const { otherSessions } = useMultipleSessions({ user });
+  const { navigateAfterSignOut, navigateAfterMultiSessionSingleSignOutUrl } = useSignOutContext();
+
+  const handleSignOut = () => {
+    if (otherSessions.length === 0) {
+      return signOut(navigateAfterSignOut);
+    }
+
+    return signOut(navigateAfterMultiSessionSingleSignOutUrl, { sessionId: session?.id });
+  };
+
+  const isLoading = userMemberships?.isLoading || userInvitations?.isLoading || userSuggestions?.isLoading;
+  const hasExistingResources = !!(userMemberships?.count || userInvitations?.count || userSuggestions?.count);
+  const identifier = user?.primaryEmailAddress?.emailAddress ?? user?.username;
+
+  return (
+    <Flow.Root flow='taskSelectOrganization'>
+      <Card.Root>
+        <Card.Content sx={t => ({ padding: `${t.space.$8} ${t.space.$none} ${t.space.$none}`, gap: t.space.$6 })}>
+          {isLoading ? (
+            <Flex
+              direction={'row'}
+              align={'center'}
+              justify={'center'}
+              sx={t => ({
+                height: '100%',
+                minHeight: t.sizes.$100,
+              })}
+            >
+              <Spinner
+                size={'lg'}
+                colorScheme={'primary'}
+                elementDescriptor={descriptors.spinner}
+              />
+            </Flex>
+          ) : (
+            <>
+              <Header.Root
+                showLogo
+                sx={t => ({ padding: `${t.space.$none} ${t.space.$8}` })}
+              >
+                <Header.Title localizationKey={localizationKeys('taskSelectOrganization.title')} />
+                <Header.Subtitle localizationKey={localizationKeys('taskSelectOrganization.subtitle')} />
+              </Header.Root>
+
+              <TaskSelectOrganizationFlows initialFlow={hasExistingResources ? 'select' : 'create'} />
+            </>
+          )}
+        </Card.Content>
+
+        <Card.Footer>
+          <Card.Action elementId='signOut'>
+            {identifier && (
+              <Card.ActionText
+                localizationKey={localizationKeys('taskSelectOrganization.signOut.actionText', {
+                  identifier,
+                })}
+              />
+            )}
```

Comment:
We should apply a text truncate here - I've got this issue while running integration tests and the email was quite large:

<img width="401" height="72" alt="image" src="https://github.com/user-attachments/assets/5b95195a-623a-45dc-bdb1-2e9f0e1bcf8b" />